src/zenml/models/base_models.py

```diff
@@ -47,40 +62,80 @@ def __eq__(self, other: Any) -> bool:
         Returns:
             True if the other object is of the same type and has the same UUID.
         """
-        return self.id == other.id if isinstance(other, DomainModel) else False
-
-    id: UUID = Field(default_factory=uuid4, title="The unique resource id.")
-    created: datetime = Field(
-        default_factory=datetime.now,
-        title="Time when this resource was created.",
-    )
-    updated: datetime = Field(
-        default_factory=datetime.now,
-        title="Time when this resource was last updated.",
-    )
+        if isinstance(other, BaseResponseModel):
+            return self.id == other.id
+        else:
+            return False
 
 
-class UserOwnedDomainModel(DomainModel):
+class UserScopedResponseModel(BaseResponseModel):
     """Base user-owned domain model.
 
     Used as a base class for all domain models that are "owned" by a user.
     """
 
-    user: UUID = Field(
-        title="The id of the user that created this resource.",
+    user: Optional["UserResponseModel"] = Field(
+        title="The user that created this resource.", nullable=True
     )
 
 
-class ProjectScopedDomainModel(UserOwnedDomainModel):
+class ProjectScopedResponseModel(UserScopedResponseModel):
     """Base project-scoped domain model.
 
     Used as a base class for all domain models that are project-scoped.
     """
 
+    project: "ProjectResponseModel" = Field(
+        title="The project of this resource."
+    )
+
+
+class ShareableResponseModel(ProjectScopedResponseModel):
+    """Base shareable project-scoped domain model.
+
+    Used as a base class for all domain models that are project-scoped and are
+    shareable.
+    """
+
+    is_shared: bool = Field(
+        title=(
+            "Flag describing if this resource is shared with other users in "
+            "the same project."
+        ),
+    )
+
+
+# -------------- #
+# REQUEST MODELS #
+# -------------- #
+
+
+class BaseRequestModel(AnalyticsTrackedModelMixin):
+    """Base request model.
+
+    Used as a base class for all request models.
+    """
+
+
+class UserOwnedRequestModel(BaseRequestModel):
```

Comment:
Shouldn't this be called `UserScopedRequestModel` to match the renaming of the response model ?

Reply:
Renamed it!